source/Calamari/Kubernetes/Commands/KubernetesApplyRawYamlCommand.cs

```diff
@@ -4,66 +4,77 @@
     public class KubernetesApplyRawYamlCommand : Command
     {
-        private const string Name = "kubernetes-apply-raw-yaml";
+        public const string Name = "kubernetes-apply-raw-yaml";
 
-        private readonly ILog log;
         private readonly IDeploymentJournalWriter deploymentJournalWriter;
         private readonly IVariables variables;
-        private readonly ICommandLineRunner commandLineRunner;
+        private readonly Kubectl kubectl;
+        private readonly DelegateInstallConvention.Factory delegateInstallFactory;
+        private readonly Func<SubstituteInFilesConvention> substituteInFilesFactory;
+        private readonly Func<ConfigurationTransformsConvention> configurationTransformationFactory;
+        private readonly Func<ConfigurationVariablesConvention> configurationVariablesFactory;
+        private readonly Func<StructuredConfigurationVariablesConvention> structuredConfigurationVariablesFactory;
         private readonly ICalamariFileSystem fileSystem;
         private readonly IExtractPackage extractPackage;
-        private readonly ISubstituteInFiles substituteInFiles;
-        private readonly IStructuredConfigVariablesService structuredConfigVariablesService;
-        private readonly ResourceStatusReportExecutor statusReportExecutor;
         private readonly IAwsAuthConventionFactory awsAuthConventionFactory;
+        private readonly Func<KubernetesAuthContextConvention> kubernetesAuthContextFactory;
+        private readonly Func<GatherAndApplyRawYamlConvention> gatherAndApplyRawYamlFactory;
+        private readonly Func<ResourceStatusReportConvention> resourceStatusReportFactory;
+        private readonly ConventionProcessor.Factory conventionProcessorFactory;
+        private readonly RunningDeployment.Factory runningDeploymentFactory;
 
         private PathToPackage pathToPackage;
 
         public KubernetesApplyRawYamlCommand(
-            ILog log,
             IDeploymentJournalWriter deploymentJournalWriter,
             IVariables variables,
-            ICommandLineRunner commandLineRunner,
+            Kubectl kubectl,
+            DelegateInstallConvention.Factory delegateInstallFactory,
+            Func<SubstituteInFilesConvention> substituteInFilesFactory,
+            Func<ConfigurationTransformsConvention> configurationTransformationFactory,
+            Func<ConfigurationVariablesConvention> configurationVariablesFactory,
+            Func<StructuredConfigurationVariablesConvention> structuredConfigurationVariablesFactory,
+            IAwsAuthConventionFactory awsAuthConventionFactory,
+            Func<KubernetesAuthContextConvention> kubernetesAuthContextFactory,
+            Func<GatherAndApplyRawYamlConvention> gatherAndApplyRawYamlFactory,
+            Func<ResourceStatusReportConvention> resourceStatusReportFactory,
+            ConventionProcessor.Factory conventionProcessorFactory,
+            RunningDeployment.Factory runningDeploymentFactory,
```

Comment:
I'm not sure this is an improvement - what problem are we solving by converting these all to factories?  I can't spot anywhere in your other PRs that requires this (correct me if I'm wrong).
Seeing these makes me concerned that we're either using DI for DI's sake, or we have a design issue with our command.

Reply:
After our chat this morning I've reverted the injection of the conventions as well as made a few other tidy ups. Looking pretty clean now I think.